pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -384,6 +386,155 @@ func convertGRPCRoute(r k8s.GRPCRouteRule, ctx configContext,
 	return vs, nil
 }
 
+func convertDestinationRule(r configContext) []config.Config {
+	var drs []config.Config
+
+	for _, obj := range r.BackendLBPolicy {
+		drs = append(drs, buildBackendLBPolicyDestinationRule(r, obj)...)
+	}
+
+	return drs
+}
+
+func buildBackendLBPolicyDestinationRule(r configContext, obj config.Config) []config.Config {
+	backendLBPolicy, _ := obj.Spec.(*k8salpha.BackendLBPolicySpec)
+	if backendLBPolicy.SessionPersistence == nil {
+		return nil
+	}
+	wrappedStatus, _ := obj.Status.(*kstatus.WrappedStatus)
+	if wrappedStatus.Status == nil {
+		wrappedStatus.Status = &k8salpha.PolicyStatus{}
+	}
+	reportStatus := func(results []RouteParentResult) {
+		obj.Status.(*kstatus.WrappedStatus).Mutate(func(s config.Status) config.Status {
+			rs := s.(*k8salpha.PolicyStatus)
+			rs.Ancestors = createPolicyStatus(results, obj, rs.Ancestors)
+			return rs
+		})
+	}
+
+	var err error
+	var drs []config.Config
+	var ancestorRefs []routeParentReference
+	for _, ref := range backendLBPolicy.TargetRefs {
+		dr := &istio.DestinationRule{
+			Host: fmt.Sprintf("%s.%s.svc.%s", ref.Name, obj.Namespace, r.Domain),
```

Comment:
I think this is incorrectly assuming the targetRef MUST be a Service. Or is that verified in some other way?

Reply:
Yep you are right. Added some logic to only accept Service targets.